cli/validate_inputs.go

```diff
@@ -0,0 +1,317 @@
+// variables from extra_arguments blocks to see if there are any TF_VAR environment variables that set terraform
+// variables. This will return the list of names of variables that are set in this way by the given terragrunt
+// configuration.
+func getTerraformInputNamesFromEnvVar(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) []string {
+	envVars := terragruntOptions.Env
+
+	// Make sure to check if there are configured env vars in the parsed terragrunt config.
+	if terragruntConfig.Terraform != nil {
+		for _, arg := range terragruntConfig.Terraform.ExtraArgs {
+			// only focus on plan args
+			if arg.EnvVars != nil && util.ListContainsElement(arg.Commands, "plan") {
+				for key, val := range *arg.EnvVars {
+					envVars[key] = val
+				}
+			}
+		}
+	}
+
+	out := []string{}
+	for envName, _ := range envVars {
+		if strings.HasPrefix(envName, TFVarPrefix) {
+			out = append(out, strings.TrimPrefix(envName, fmt.Sprintf("%s_", TFVarPrefix)))
+		}
+	}
+	return out
+}
+
+// getTerraformInputNamesFromConfig will return the list of names of variables configured by the inputs block in the
+// terragrunt config.
+func getTerraformInputNamesFromConfig(terragruntConfig *config.TerragruntConfig) []string {
+	out := []string{}
+	for inputName, _ := range terragruntConfig.Inputs {
+		out = append(out, inputName)
+	}
+	return out
+}
+
+// getTerraformInputNamesFromVarFiles will return the list of names of variables configured by var files set in the
+// extra_arguments block of the given terragrunt config.
+func getTerraformInputNamesFromVarFiles(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) ([]string, error) {
+	if terragruntConfig.Terraform == nil {
+		return nil, nil
+	}
+
+	varFiles := []string{}
+	for _, arg := range terragruntConfig.Terraform.ExtraArgs {
+		// only focus on plan args
+		if util.ListContainsElement(arg.Commands, "plan") {
+			varFiles = append(varFiles, arg.GetVarFiles(terragruntOptions.Logger)...)
+		}
+	}
+
+	return getVarNamesFromVarFiles(varFiles)
+}
+
+// getTerraformInputNamesFromCLIArgs will return the list of names of variables configured by -var and -var-file CLI
+// args that are passed in via the configured arguments attribute in the extra_arguments block of the given terragrunt
+// config and those that are directly passed in via the CLI.
+func getTerraformInputNamesFromCLIArgs(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) ([]string, error) {
+	inputNames, varFiles, err := getVarFlagsFromArgList(terragruntOptions.TerraformCliArgs)
+	if err != nil {
+		return inputNames, err
+	}
+
+	if terragruntConfig.Terraform != nil {
+		for _, arg := range terragruntConfig.Terraform.ExtraArgs {
+			// only focus on plan args
+			if util.ListContainsElement(arg.Commands, "plan") {
+				if arg.Arguments != nil {
+					vars, rawVarFiles, err := getVarFlagsFromArgList(*arg.Arguments)
```

Comment:
Is this different than what `getTerraformInputNamesFromVarFiles` is doing?

Reply:
Yes this is different. `getTerraformInputNamesFromVarFiles` is extracting var files from `required_var_files` and `optional_var_files`, while this is extracting it from the raw args list.

Updated comments to clarify this: 70edaf281643441cbfa489b023c9b6cf1d078122